File: llm-python/security/channel_guard/service.py
class ChannelGuardService:
    def __init__(self, enforce: bool = False):
        self._enforce = enforce

    def validate_source(self, headers: dict[str, str]) -> None:
        if not self._enforce:
            return
        if (headers.get("x-trusted-channel") or "").strip() != "vpn":
            raise PermissionError("untrusted source channel")

File: portal-go/internal/security/channelguard/service.go
package channelguard

import (
	"context"
	"errors"
	"net/http"
	"strings"
)

type Service struct {
	enforce bool
}

func New(enforce bool) *Service {
	return &Service{enforce: enforce}
}

func (s *Service) ValidateSource(_ context.Context, r *http.Request) error {
	if !s.enforce {
		return nil
	}
	if strings.TrimSpace(r.Header.Get("X-Trusted-Channel")) != "vpn" {
		return errors.New("untrusted source channel")
	}
	return nil
}
